lager/any_lens.hpp

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <lager/lenses.hpp>
+#include <memory>
+#include <type_traits>
+#include <utility>
+
+namespace lager {
+namespace lens {
+namespace detail {
+
+template <typename Whole, typename Part>
+struct lens_i {
+    virtual Part view(Whole const&) const              = 0;
+    virtual Whole set(Whole const&, Part const&) const = 0;
+};
+
+template <typename Lens, typename Whole, typename Part>
+struct lens_holder : public lens_i<Whole, Part> {
+    Lens value;
+
+    template <typename T>
+    lens_holder(T&& other) : value{std::forward<T>(other)} {}
+
+    Part view(Whole const& w) const override { return ::lager::view(value, w); }
+
+    Whole set(Whole const& w, Part const& p) const override {
+        return ::lager::set(value, w, p);
+    }
+};
+
+} // namespace detail
+
+template <typename Whole, typename Part>
+class any_lens : zug::detail::pipeable {
```

Comment:
agreed with `lens<>` instead of `any_lens<>`, the main motivation was to discourage its use over non type erased alternatives.
Not sure about the name collision problem, I'm kind of concerned about introducing too many breaking changes.